File: backend/src/main/java/com/github/denvned/xodus/compare/backend/graphql/AddedEntityConnection.java
package com.github.denvned.xodus.compare.backend.graphql;

import com.github.denvned.graphql.annotations.GraphQLField;
import com.github.denvned.graphql.annotations.GraphQLNonNull;
import jetbrains.exodus.entitystore.EntityIterable;

import java.util.AbstractList;
import java.util.List;

public final class AddedEntityConnection extends AbstractEntityBasedConnection {
    public AddedEntityConnection(EntityIterable entities, Integer first, Long after) {
        super(entities, first, after);
    }

    @GraphQLField @GraphQLNonNull
    public List<@GraphQLNonNull AddedEntityEdge> getEdges() {
        init();

        return new AbstractList<AddedEntityEdge>() {
            @Override
            public AddedEntityEdge get(int index) {
                return new AddedEntityEdge(entities.get(index));
            }

            @Override
            public int size() {
                return entities.size();
            }
        };
    }
}

File: backend/src/main/kotlin/com/github/denvned/xodus/compare/backend/graphql/AddedEntityConnection.kt
package com.github.denvned.xodus.compare.backend.graphql

import jetbrains.exodus.entitystore.EntityIterable
import java.util.*

class AddedEntityConnection(entities: EntityIterable, first: Int?, after: Long?)
    : AbstractEntityBasedConnection(entities, first, after) {

  val edges: List<AddedEntityEdge>
    get() = object : AbstractList<AddedEntityEdge>() {
      override val size get() = page.entities.size
      override fun get(index: Int) = AddedEntityEdge(page.entities[index])
    }
}
